tests/accessor/accessor_default_values.h

```diff
@@ -0,0 +1,174 @@
+
+ public:
+  void operator()(const std::string& type_name,
+                  const std::string& access_mode_name,
+                  const std::string& target_name) {
+    auto section_name =
+        get_section_name<Dimension>(type_name, access_mode_name, target_name,
+                                    "Verify default value for dimensions "
+                                    "template parameter. [generic accessor]");
+    SECTION(section_name) {
+      REQUIRE(std::is_same_v<sycl::accessor<T>,
+                             sycl::accessor<T, detail::expected_dims>>);
+    }
+    section_name =
+        get_section_name<Dimension>(type_name, access_mode_name, target_name,
+                                    "Verify default value for accessMode "
+                                    "template parameter. [generic accessor]");
+    SECTION(section_name) {
+      REQUIRE(std::is_same_v<
+              sycl::accessor<T, Dimension>,
+              sycl::accessor<T, Dimension, detail::expected_mode<T>>>);
+    }
+    section_name =
+        get_section_name<Dimension>(type_name, access_mode_name, target_name,
+                                    "Verify default value for accessTarget "
+                                    "template parameter. [generic accessor]");
+    SECTION(section_name) {
+      REQUIRE(std::is_same_v<sycl::accessor<T, Dimension, AccessMode>,
+                             sycl::accessor<T, Dimension, AccessMode,
+                                            detail::expected_target>>);
+    }
+    section_name =
+        get_section_name<Dimension>(type_name, access_mode_name, target_name,
+                                    "Verify default value for isPlaceholder "
+                                    "template parameter. [generic accessor]");
+    SECTION(section_name) {
+      REQUIRE(std::is_same_v<sycl::accessor<T, Dimension, AccessMode, Target>,
+                             sycl::accessor<T, Dimension, AccessMode, Target,
+                                            detail::expected_placeholder>>);
+    }
+  }
+};
+
+/**
+ * @brief Struct with functor that will be used in type coverage to call all
+ *        verification functions.
+ * @tparam T Current data type
+ * @param type_name Current data type string representation
+ */
+template <typename T>
+class run_tests {
+ public:
+  void operator()(const std::string& type_name) {
+    // Type packs instances have to be const, otherwise for_all_combination
+    // will not compile
+    const auto access_modes = get_access_modes();
+    const auto dimensions = get_all_dimensions();
+    const auto targets = get_targets();
+
+    // Run test with non const data type
+    for_all_combinations<test_for_generic_acc, T>(access_modes, dimensions,
+                                                  targets, type_name);
+    for_all_combinations<test_for_host_acc, T>(access_modes, dimensions,
+                                               targets, type_name);
```

Comment:
Looks like we don't need to iterate over `access_modes` and `targets` to check default template parameters of `host_accessor`

Reply:
this comment has been applied in this commit: https://github.com/KhronosGroup/SYCL-CTS/pull/335/commits/fdc3712565b19f017f6c64f03c15c0edfeaf25e0